Add Live Preview activation event

okmttdhr opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to https://github.com/Automattic/dotcom-forge/issues/3301

## Proposed Changes

This PR is a follow-up for #80953 and https://github.com/Automattic/wp-calypso/pull/80956.
Add an event that is fired when a user clicks `Activate` or `Activate & Save` when previewing a theme.

wpcom_block_editor_save_click
<img width="750" alt="Screen Shot 2023-08-24 at 14 19 40" src="https://github.com/Automattic/wp-calypso/assets/5287479/be92a711-832f-4cac-9b54-382a47bd8a06">

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Patch this PR to your sandbox with `cd apps/wpcom-block-editor/ && yarn dev --sync`
* Sandbox widgets.wp.com and your site
* Go to the Site Editor
* See if events fire with https://github.com/Automattic/tracks-chrome-extension

UI that triggers the event

[![Image from Gyazo](https://i.gyazo.com/b59c9aad5a0e3854810f7caf60d62c05.gif)](https://gyazo.com/b59c9aad5a0e3854810f7caf60d62c05)

[![Image from Gyazo](https://i.gyazo.com/6c580580b87a002e44386fcfc11265d1.gif)](https://gyazo.com/6c580580b87a002e44386fcfc11265d1)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

okmttdhr committed:
Add the activation event when previewing

taipeicoder commented on apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-editor-save-click.js:
```diff
@@ -12,14 +12,18 @@ export const wpcomBlockEditorSaveClick = () => ( {
 
 		if ( isSiteEditor ) {
-			actionType = 'save';
+			if ( event.target.textContent.includes( 'Activate' ) ) {
```
We need to factor in different translations so we can't depend on the text being in English.

okmttdhr replied:
I had a try to address this 🙏 https://github.com/Automattic/wp-calypso/pull/81000/commits/a7870dde524001f07c05fca8a5dd0d902e52a8b9

okmttdhr committed:
Don't depend on the text

arthur791004 commented on apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-block-editor-save-click.js:
```diff
@@ -19,7 +19,12 @@ export const wpcomBlockEditorSaveClick = () => ( {
 
 		if ( isSiteEditor ) {
-			actionType = 'save';
+			const isPreviewingBlockTheme = window.location.search.includes( 'wp_theme_preview' );
```
It might be better to use [isPreviewingTheme](https://github.com/WordPress/gutenberg/blob/12f305ece929393fc246544b03f7881e8a0923a1/packages/edit-site/src/utils/is-previewing-theme.js#L6) if possible. Otherwise, we can simply copy & paste the implementation to here as using `includes` might not be very accurate 😅

okmttdhr replied:
I had a go for another way; https://github.com/Automattic/wp-calypso/pull/81000/commits/705c9324a03860533b180d3d4bd05cc1d4cd7313

okmttdhr committed:
Use the text with translation for a selector

Use the same way as https://github.com/Automattic/wp-calypso/pull/81002#discussion_r1305230361

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8814824

Thank you @okmttdhr for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8814824).
